crates/uv-distribution/src/metadata/requires_dist.rs

```diff
@@ -226,6 +226,29 @@ mod test {
         "###);
     }
 
+    #[tokio::test]
+    async fn wrong_type() {
+        let input = indoc! {r#"
+            [project]
+            name = "foo"
+            version = "0.0.0"
+            dependencies = [
+              "tqdm",
+            ]
+            [tool.uv.sources]
+            tqdm = true
+        "#};
+
+        assert_snapshot!(format_err(input).await, @r###"
+        error: TOML parse error at line 8, column 8
+          |
+        8 | tqdm = true
+          |        ^^^^
+        invalid type: boolean `true`, expected an array or map
```

Comment:
This is the payoff. Without this change, you just get: `data did not match any variant of untagged enum SourcesWire`